include/pmacc/memory/buffers/HostBufferIntern.hpp

```diff
@@ -124,12 +124,12 @@ class HostBufferIntern : public HostBuffer<TYPE, DIM>
     void setValue(const TYPE& value)
     {
         __startOperation(ITask::TASK_HOST);
-        size_t current_size = this->getCurrentSize();
+        long current_size = static_cast< long >(this->getCurrentSize());
```

Comment:
please use `int64_t` instead of long because the size of  `long` is not defined and depend on the compiler. If it should be 32 bit please use `int32_t` or `int` (even if int is also not defined)

Reply:
Ok. Whether it should be 32 bit or 64 bit I'm not sure. It was size_t originally, so I guess 64 bit?

Reply:
Yes, since this is executed on the CPU, memory is usually not a constrain. Using 64 bits would be saver.

Reply:
This is not the point here.
The `getCurrentSize()` is the number of elements of a container and can potentially be large even on node.
Ideally, we should always assume `int64_t` for such elements, 32bit can be too small in some cases (e.g. 2 billion floats of a particle attribute is just 7.8 GByte)